### PR TITLE
Restructure project

### PR DESCRIPTION
**Pull request by Hornwitser**
Implementation of the package structure proposed in #316 with the exception of plugins and support for loading plugins from node packages.  To be consistent with the other binaries clusterctl is renamed to clusterioctl.

### CODE REVIEW

**Review by psihius — APPROVED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -6,7 +6,8 @@ before_install:
   - tar -xf factorio.tar.gz
   - npm install -g codecov
 script:
-  - node build-mod --bump-patch --factorio-version 0.18 --output-dir sharedMods
+  - npx lerna bootstrap --no-ci
```
Comment (Danielv123) [inline .travis.yml:9]: Why would you use --no-ci on CI?

File: README.md [hunk 2]
```diff
@@ -105,12 +105,13 @@ supported on EOL Ubuntu releases so make sure you're on a recent release of Ubun
...
 
     wget -qO - https://deb.nodesource.com/setup_12.x | sudo -E bash -
-    sudo apt install -y nodejs python-dev git build-essential
-    git clone -b master https://github.com/clusterio/factorioClusterio.git
-    cd factorioClusterio
+    sudo apt install -y nodejs
+    mkdir clusterio
+    cd clusterio
+    npm init -y
+    npm install @clusterio/master @clusterio/slave @clusterio/ctl
```
Comment (Danielv123) [inline README.md:112]: Possibly use npm init @clusterio ?

File: README.md [hunk 3]
```diff
@@ -136,30 +137,19 @@ The -v flag is used to specify the instance directory. Your instances (save file
...
 
 1. Open PowerShell or Command prompt in the directory you want to install to and run the following commands.
 
-        git clone -b master https://github.com/clusterio/factorioClusterio
-        cd factorioClusterio
-        npm install --only=production
+        mkdir clusterio
+        cd clusterio
+        npm init -y
+        npm install @clusterio/master @clusterio/slave @clusterio/ctl
```
Comment (Danielv123) [inline README.md:152]: Same here ^

File: docs/communication.md [hunk 4]
```diff
@@ -49,9 +49,10 @@ Message
 
 There's two types of messages currently implemented apart from the
 Link builtin close message and that's Request and Event.  The builtin
-messages are defined in [lib/link/messages.js](../lib/link/messages.js)
-and are attached to links via the attachAllMessages function, which is
-usally called in the Link subclass constructor.
+messages are defined in
+[lib/link/messages.js](/package/lib/link/messages.js) and are attached
```
Comment (Danielv123) [inline docs/communication.md:53]: Path is updated, text is not?

File: docs/config-system.md [hunk 5]
```diff
@@ -6,15 +6,15 @@ configuration system in Clusterio, and unless you're developing
...
+the master server, instances, slaves, and the clusterioctl utility.
 Plugins get their own config group for the master server and instances
 which they can add their own fields to.  The fields are defined early in
 the startup, and the groups and their fields for disabled plugins will
 still be created.
 
 The built-in config groups and their fields are defined in
-[/lib/config/definitions.js](lib/config/definitions.js) while plugins
-can add their own config groups to the `MasterConfigGroup` and
+[lib/config/definitions.js](/packages/lib/config/definitions.js) while
```
Comment (Danielv123) [inline docs/config-system.md:16]: Same here. This is developer documentation, I don't think its a given that a new developer would know to look in /packages

**Review by Danielv123 — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (Danielv123) [reply README.md:112]: You will have to decide if you want to add that here or later. Either makes sense.

File: .travis.yml (hunk 1 above)
Comment (Hornwitser) [reply .travis.yml:9]: As noted in the commit that does that it reqires package-lock.json files to be commited to the repository and we do not for the time being do this.

File: README.md (hunk 2 above)
Comment (Hornwitser) [reply README.md:112]: Can't do that until the `@clusterio/create` exists, which it doesn't.  This is a method that can be used until such a package is created.

Comment (Hornwitser) [reply README.md:112]: To reiterate, the documentation should reflect the current state of the project.  Not some future idea that may or may not have been implemented.  If the create package is implemented then the documentation here will be changed to reflect that.

File: .travis.yml (hunk 1 above)
Comment (Danielv123) [reply .travis.yml:9]: Makes sense, we should maybe consider doing that sometime, but that doesn't belong here.

**Review by Danielv123 — APPROVED** [review]

**Pull request merged**